### ISSUE TITLE
Godspeed, you magnificent maniac...

### ISSUE THREAD

**Issue by vt-idiot** [Merramore/stable-diffusion-webui-extension-DirectML#1]
Is this what I think it is? An extension to make AUTOMATIC1111 webui run using [pytorch-directml](https://pypi.org/project/pytorch-directml/), _and thus potentially on AMD GPUs in Windows instead of under Linux with ROCm and a million asterisks_?

Godspeed. I took one look at making it work and decided it wasn't worth the headache of replacing every single `torch...cuda` call with `torch...dml` only to later find that something other than pytorch was also CUDA only and getting in my way, or that a function wasn't supported.

...I'll start a bounty at $42.0

**Comment by MrNehr** [comment]
It work really well with my RX 5600 XT. Just found out about this. It's a life saver. From 5min on CPU to 32sec.

**Comment by vt-idiot** [comment]
> It work really well with my RX 5600 XT. Just found out about this. It's a life saver. From 5min on CPU to 32sec.

You got stable-diffusion-webui running on Windows with an AMD GPU? How?

**Comment by MrNehr** [comment]
Using this extension and the automatic1111 SD.

**Comment by vt-idiot** [comment]
> Using this extension and the automatic1111 SD.

it _already_ works as is?! there goes my weekend

**Comment by vt-idiot** [comment]
bah. it works. kind of. instant `VIDEO_MEMORY_MANAGEMENT_INTERNAL` BSOD when I try to generate anything with an 8GB RX5700

**Comment by MrNehr** [comment]
I don't know if it's related but i'm using this driver: [https://www.amd.com/en/support/kb/release-notes/rn-rad-win-22-11-1-mlir-iree](https://www.amd.com/en/support/kb/release-notes/rn-rad-win-22-11-1-mlir-iree)

**Comment by vt-idiot** [comment]
Whelp, there's an experiment for next weekend. It might be related.

What resolution do you usually generate at? I went straight for 768x512, didn't launch with --medvram or anything like that either.

**Comment by MrNehr** [comment]
I do 512x512, but i did some 512x816 and it was fine.

edit: you might still need --precision full --no-half (i do have them)

**Comment by vt-idiot** [comment]
> edit: you might still need --precision full --no-half (i do have them)

Yeah I went for `--skip-torch-cuda-test --precision full --no-half --no-half-vae` to begin with. I'll try it again later with those drivers you recommended and start out smaller.

Does it work on all samplers, or rather, have you used it with particular samplers? DPM++ SDE Karras didn't work (but didn't BSOD) because of the way it tries to access the device. IIRC DPM++ 2s a Karras was what I tried next and that caused the BSOD.

**Comment by MrNehr** [comment]
Yeah some sampler don't work. Euler a, Euler, Heun, DDIM, DPM++ 2S a, DPM++ 2S a karra work fine.
I didn't test them all but i do know some DPM don't work for me.
